### PR TITLE
refactor: prepare CreateGroupConversationActionHandler for async WBP-5136

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

- Wrap the entire response handler in `CreateGroupConversationActionHandler` inside a task in preparation for methods turning async.
- Inject the `MLSService` instead of looking it up on the context.

NOTE: these changes generated some churn on the tests since they relied on that all `ActionHandler` instances are only instantiated with a `NSManagedContext`.

Needs releases with:

- [ ] https://github.com/wireapp/wire-ios/pull/759

#### Test Coverage

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/CreateGroupConversationActionHandler.swift [hunk 1]
```diff
@@ -185,86 +195,77 @@ final class CreateGroupConversationActionHandler: ActionHandler<CreateGroupConve
...
-            let newConversation = processor.updateOrCreateConversation(
-                from: payload,
-                in: context
-            )
-        else {
-            Logging.network.warn("Can't process response, aborting.")
-            action.fail(with: .proccessingError)
-            return
-        }
+        Task {
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/CreateGroupConversationActionHandler.swift]: I thought we were going to turn this async directly? But i supposed we only need it in this action handler, right?

What about cancelling the task if we're cancelling/removing the action

Comment (typfel) [reply wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/CreateGroupConversationActionHandler.swift]: How can I turn it async directly? I'm all for it if it's possible.

> What about cancelling the task if we're cancelling/removing the action

What do you mean? This is the function which handles the response from the backend. How can it be cancelled?

Comment (netbe) [reply wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/CreateGroupConversationActionHandler.swift]: yes right, I was thinking if we cancel the request, the handleResponse should not be processed.

For tuning it async, see here it's one level https://github.com/wireapp/wire-ios/pull/766/files#diff-32f98bd1aa3fcf653f5cab8d131d223dd21f61368dff0b71a1ce49979a005b15R117

**Review by netbe — APPROVED** [review]

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/CreateGroupConversationActionHandler.swift (hunk 1 above)
Comment (typfel) [reply wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/CreateGroupConversationActionHandler.swift]: Ok, yeah that's nicer

**Review by caldrian — APPROVED** [review]

**Pull request merged**